File: Problems/0226. (E) Invert Binary Tree/Python3/solution.py
class Solution:
  def invertTree(self, root: Optional[TreeNode]) -> Optional[TreeNode]:
    # recursively swap subtrees
    
    if root is not None:
      left = self.invertTree(root.right)
      right = self.invertTree(root.left)
      root.left, root.right = left, right
    return root

File: Problems/0226. (E) Invert Binary Tree/Java/solution.java
class Solution {
  public TreeNode invertTree(TreeNode root) {
    // recursively swap subtrees
    if (root != null) {
      TreeNode left = invertTree(root.right);
      TreeNode right = invertTree(root.left);
      root.left = left;
      root.right = right;
    }
    return root;
  }
}
